pkg/users/git.go

```diff
@@ -60,6 +60,9 @@ func (r *GitUserResolver) GitUserSliceAsUserDetailsSlice(users []gits.GitUser) (
 // * making a call to the gitProvider
 // as often user info is not complete in a git response
 func (r *GitUserResolver) Resolve(user *gits.GitUser) (*jenkinsv1.User, error) {
+	user.Login = naming.ToValidValue(user.Login)
+	user.Login = naming.ToValidName(user.Login)
```

Comment:
Also, this change breaks an existing test in git_test.go. Not sure whether this is the right way to do it.
You also risk a panic now. The _nil_ check for user comes below this assignment, so for a _nil_ user you will get a panic.

Reply:
No those are two different things we do the user login, that's always needed. I think if we first sanitized the users login, we could then use it below as we please.

Reply:
Hmm, it still breaks something though

Reply:
I see now that these re two different calls. My mistake. Looks very similar :-(. The panic issue remains and I have not checked whether this is conceptually right

Reply:
actually this is not at all what you want. You pulled the modification which otherwise happens in the _selectUsers_ function into Resolve as well, changing the behaviour completely.

_selectUsers_ is a function declaration and the function gets passed as parameter. Don't ask me why. IMO it does not make any sense at all at this point to use function pointers and it just obfuscates the code.